pkg/service/service.go

```diff
@@ -219,6 +221,76 @@ func ListWithDetailedStatus(client *occlient.Client, applicationName string) (Se
 	}, nil
 }
 
+// ListOperatorServices lists all operator backed services
+func ListOperatorServices(client *kclient.Client) ([]unstructured.Unstructured, error) {
+	klog.V(4).Info("Getting list of services")
+
+	// First let's get the list of all the operators in the namespace
+	csvs, err := client.GetClusterServiceVersionList()
+	if err != nil {
+		return nil, errors.Wrap(err, "Unable to list operator backed services")
+	}
+
+	var allCRInstances []unstructured.Unstructured
+
+	// let's get the Services a.k.a Custom Resources (CR) defined by each operator, one by one
+	for _, csv := range csvs.Items {
+		klog.V(4).Infof("Getting services started from operator: %s\n", csv.Name)
+		customResources := client.GetCustomResourcesFromCSV(csv)
+
+		// list and write active instances of each service/CR
+		instances, err := getCRInstances(client, customResources)
+		if err != nil {
+			return nil, err
+		}
+
+		// assuming there are more than one instances of a CR
+		allCRInstances = append(allCRInstances, instances...)
+	}
+
+	return allCRInstances, nil
+}
+
+// getGVRFromCR parses and returns the values for group, version and resource
+// for a given Custom Resource (CR).
+func getGVRFromCR(cr olm.CRDDescription) (group, version, resource string, err error) {
+	version = cr.Version
+
+	gr := strings.SplitN(cr.Name, ".", 2)
+	if len(gr) != 2 {
+		err = fmt.Errorf("Couldn't split Custom Resource's name into two: %s\n", cr.Name)
+		return
+	}
+	resource = gr[0]
+	group = gr[1]
+
+	return
+}
+
+// getCRInstances returns active instances of given Custom Resource (service in
+// odo lingo) in the active namespace of the cluster
+func getCRInstances(client *kclient.Client, customResources []olm.CRDDescription) ([]unstructured.Unstructured, error) {
+	var instances []unstructured.Unstructured
+
+	for _, cr := range customResources {
+		klog.V(4).Infof("Getting instances of: %s\n", cr.Name)
+		group, version, resource, err := getGVRFromCR(cr)
+		if err != nil {
+			return []unstructured.Unstructured{}, err
```

Comment:
why not just return nil?

Reply:
So that it doesn't crash client code if error is not checked… returning an empty collection makes sense.